mteb/tasks/STS/multilingual/IndicCrosslingualSTS.py

```diff
@@ -0,0 +1,90 @@
+from __future__ import annotations
+
+import datasets
+
+from mteb.abstasks.TaskMetadata import TaskMetadata
+
+from ....abstasks import AbsTaskSTS, MultilingualTask
+
+_LANGUAGES = {
+    "en-as": ["eng-Latn", "asm-Beng"],
+    "en-bn": ["eng-Latn", "ben-Beng"],
+    "en-gu": ["eng-Latn", "guj-Gujr"],
+    "en-hi": ["eng-Latn", "hin-Deva"],
+    "en-kn": ["eng-Latn", "kan-Knda"],
+    "en-ml": ["eng-Latn", "mal-Mlym"],
+    "en-mr": ["eng-Latn", "mar-Deva"],
+    "en-or": ["eng-Latn", "ory-Orya"],
+    "en-pa": ["eng-Latn", "pan-Guru"],
+    "en-ta": ["eng-Latn", "tam-Taml"],
+    "en-te": ["eng-Latn", "tel-Telu"],
+    "en-ur": ["eng-Latn", "urd-Arab"],
+}
+
+
+class IndicCrosslingualSTS(AbsTaskSTS, MultilingualTask):
+    metadata = TaskMetadata(
+        name="IndicCrosslingualSTS",
+        dataset={
+            "path": "jaygala24/indic_sts",
+            "revision": "16abc16bea9e38262a8a3a74fd71ce2da51a5c3b",
+        },
+        description="This is a Semantic Textual Similarity testset between English and 12 high-resource Indic languages.",
+        reference="https://huggingface.co/datasets/jaygala24/indic_sts",
+        type="STS",
+        category="s2s",
+        eval_splits=["test"],
+        eval_langs=_LANGUAGES,
+        main_score="cosine_spearman",
+        date=("2021-04-30", "2021-06-09"),
+        form=["written", "spoken"],
+        domains=["News", "Non-fiction", "Web", "Spoken", "Government"],
+        task_subtypes=[],
+        license="CC0",
+        socioeconomic_status="mixed",
+        annotations_creators="expert-annotated",
+        dialect=[],
+        text_creation="created",
+        bibtex_citation="""@article{10.1162/tacl_a_00452,
+    author = {Ramesh, Gowtham and Doddapaneni, Sumanth and Bheemaraj, Aravinth and Jobanputra, Mayank and AK, Raghavan and Sharma, Ajitesh and Sahoo, Sujit and Diddee, Harshita and J, Mahalakshmi and Kakwani, Divyanshu and Kumar, Navneet and Pradeep, Aswin and Nagaraj, Srihari and Deepak, Kumar and Raghavan, Vivek and Kunchukuttan, Anoop and Kumar, Pratyush and Khapra, Mitesh Shantadevi},
+    title = "{Samanantar: The Largest Publicly Available Parallel Corpora Collection for 11 Indic Languages}",
+    journal = {Transactions of the Association for Computational Linguistics},
+    volume = {10},
+    pages = {145-162},
+    year = {2022},
+    month = {02},
+    issn = {2307-387X},
+    doi = {10.1162/tacl_a_00452},
+    url = {https://doi.org/10.1162/tacl\_a\_00452},
+    eprint = {https://direct.mit.edu/tacl/article-pdf/doi/10.1162/tacl\_a\_00452/1987010/tacl\_a\_00452.pdf},
+}""",
+        n_samples={"test": 10020},
```

Comment:
This becomes quite a large dataset. It might be worth reducing the size of the splits to 256 pr. language (assuming it does not influence the scores too much, which I don't believe it will)

Reply:
Yes, I acknowledge that the split size is higher but it doesn't take much time in evaluation. Both the models mentioned above took nearly 9 secs for evaluations. I will evaluate the models with the split size of 256 per language and check if the results are consistent with the larger splits.

Reply:
The split size of 10000 is across 12 Indic languages. The split size for each language is roughly around 800 which is well within the mentioned limit of 2048. I have updated the metadata for the dataset in the latest commit.

Reply:
2048 is a suggested limit no hard rules there. We are trying to keep MMTEB as small as possible (but naturally as big as required).

It is quite easy to experiment with downsampling in the `dataset_transform()`, then it will be easy enough to empirically estimate whether downsampling notably influences performance (STS tasks are quite robust to downsampling as far as I have seen).

Reply:
I have updated the dataset class to downsample the test set size of 256 per language with a stratified split. I don't see a significant drop in performance with a variance of 0.02 in correlation across a few languages.